virtcontainers/qemu_amd64.go

```diff
@@ -32,7 +32,7 @@ var qemuPaths = map[string]string{
 
 var kernelRootParams = []Param{
 	{"root", "/dev/pmem0p1"},
-	{"rootflags", "dax,data=ordered,errors=remount-ro rw"},
+	{"rootflags", "dax,data=ordered,errors=remount-ro ro"},
```

Comment:
I thought our rootfs was already read-only? Was I the only one thinking this?

Reply:
I was thinking 'why did we not do this already, forever - maybe there is a reason...' :-)
@jodh-intel - I don't suppose you remember from wayback do you?
I suspect @gnawux @bergwolf might have some thoughts as well.

Reply:
I'm frankly surprised this wasn't raised on one of the https://github.com/clearcontainers repos but cannot find any mention of it.

After a bit of digital archaeology, I found a very old comment from a test tool I wrote a few years ago that stated we needed to mount the rootfs `rw` with DAX/NVDIMM to avoid a kernel panic like this:

```
Kernel panic - not syncing: Requested init /usr/lib/systemd/systemd failed (error -14)
```

I believe we were seeing that behaviour with a `linux-container-4.2.5-44` kernel but cannot be 100% sure. 

However, since this PR does indeed work, something changed :smile:. It would be good to get input from the DAX experts on this PR.

Reply:
We even mounted `rw` in the predecessor to `clearcontainers`, `cc-oci-runtime`:

- https://github.com/intel/cc-oci-runtime/blob/master/data/kernel-cmdline

Reply:
@djbw - any thoughts from the DAX side wrt mounting ro instead of rw

Reply:
works for me with DAX enabled

```
[    0.282411] EXT4-fs (pmem0p1): DAX enabled. Warning: EXPERIMENTAL, use at your own risk
[    0.282687] EXT4-fs (pmem0p1): mounted filesystem with ordered data mode. Opts: dax,data=ordered,errors=remount-ro
```

Reply:
Yes pmem metadata simply need to be read, so no issues from this perspective.